rstudio/rstudio
# Prevent terminal line spacing from going below 1.0

### Intent

- Fixes #7827

Make sure terminal line spacing doesn't go below 1.0 (which can cause lines to start overlapping). Noticed this happening on one of my machines (Windows laptop) when I was trying latest version of xterm.js and it started throwing an error if line spacing was below 1.0. I don't know exactly what combination of display hardware, scaling, etc. is needed to get into this scenario.

### Approach

When computing line height, don't let it go below 1.0! Also, upgraded to latest xterm.js since it it kind enough to make it very obvious when this is happening via additional runtime validation checks (which might reveal other issues for us).

### QA Notes

Just try the example in the issue; there should be visible space between the lines with the fix in place.

Again, don't know exactly how you can repro this; perhaps you'll be lucky and it happens on your friendly neighborhood Windows machine, too, then you can do a before/after check.
